shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPlugin.mm

```diff
@@ -0,0 +1,169 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPlugin.h"
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+#import "flutter/fml/logging.h"
+#import "flutter/shell/platform/darwin/ios/framework/Source/FlutterViewController_Internal.h"
+
+// Method Channel name to start spell check.
+static NSString* const kInitiateSpellCheck = @"SpellCheck.initiateSpellCheck";
+
+// Spell check results will be encoded as a string representing the span of that result, with
+// the format "start_index.end_index.suggestion_1\nsuggestion_2\nsuggestion_3".
+static NSString* FormatSpellCheckResult(NSUInteger startIndex,
```

Comment:
1. Is this the same format as the android? 
2. Why are we using a custom serialization? can we just use more standardized ones like JSON or CSV?

Reply:
@camsim99 Are you worrying that JSON serialization/parsing is significantly slower (or taking more space) than the proposed custom format?

Reply:
Ah okay. I did a quick search and that seems to be true. I'm good with changing this JSON since it's standard. I'll probably get [this PR](https://github.com/flutter/flutter/pull/104460) to a landable state and then make the change in the Android engine before landing it.

cc: @justinmc

Reply:
Sounds good. I can update this PR to use JSON.

Reply:
If memory is a concern, it seems to me that we should limit the misspelled word that the API returns.

If we write our own formatter and parser, we also have to consider the performance of them, which creates an additional maintenance burden.

Also, I was a little confused myself last week. JSON serialization is already supported by default on MethodChannel, so we don't really have to do an extra serialization in this PR if we choose to use JSON. I will revert that code.

Reply:
I talked to Gary a little bit and we agreed that using JSON seems to be the way to go for now. If it turns out to be slow, we can revisit using a custom format.